app/repositories/OptOutSessionDataRepository.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import cats.data.OptionT
+import models.incomeSourceDetails.{TaxYear, UIJourneySessionData}
+import models.itsaStatus.ITSAStatus
+import models.itsaStatus.ITSAStatus.ITSAStatus
+import models.optout.OptOutSessionData
+import play.api.libs.json.{Json, OFormat}
+import repositories.OptOutContextData.{statusToString, stringToStatus}
+import services.optout.OptOutProposition
+import services.optout.OptOutProposition.createOptOutProposition
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.OptOutJourney
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class OptOutSessionDataRepository @Inject()(val repository: UIJourneySessionDataRepository) {
```

Comment:
this is all about OptOut specific operations over the repository so I think 
- it should be renamed to OptOutRepositoryService
- and moved to the services.optout package

Reply:
I see it differently. I think modules like this class should be named after the role the play in supporting the modules that use them; in this case the `OptOutService`. `OptOutService` requires journey information to be stored and retrieved, and that's what a `Repository` does.

I could be persuaded to contract the name to `OptOutRepository` or maybe `OptOutSessionRepository`, but I don't think adding `Service` helps much. I find that `Service` is over-used and is anodyne rather than  domain centric, and because of this lack of specificity,  can end up with the file being a bit of a junk-draw for random bits of code.